libopae/src/reconf.c

```diff
@@ -148,17 +161,19 @@ static fpga_result clear_port_errors(fpga_handle handle)
 		return result;
 	}
 
-	snprintf_s_ss(syfs_errpath, sizeof(syfs_errpath), "%s/%s", syfs_path, PORT_SYSFS_ERRORS);
 	// Read port error.
-	result = sysfs_read_u64(syfs_errpath, &error);
+	result = read_port_error(syfs_errpath, syfs_path, &error);
```

Comment:
`read_port_error()` modifies `sysfs_errpath`, but it's never used afterwards. From a "what does this function do" perspective, it's not clear that `read_port_error()` needs the user to pass in a buffer that then is used to construct a path that's not used again.

If you want to keep the function, you could allocate a new buffer for the resulting path inside. That way you could also be sure about its size. But as noted above, you could also just put the `snprintf_s_ss()` call here and use `sysfs_read_u64()`.